tests/parser/features/test_bytes_map_keys.py

```diff
@@ -0,0 +1,73 @@
+import pytest
+from tests.setup_transaction_tests import chain as s, tester as t, ethereum_utils as u, check_gas, \
+    get_contract_with_gas_estimation, get_contract
+from viper.exceptions import TypeMismatchException
+
+
+def test_basic_bytes_keys():
+    code = """
+mapped_bytes: num[bytes <= 5]
+
+def set(k: bytes <= 5, v: num):
+    self.mapped_bytes[k] = v
+
+
+def get(k: bytes <= 5) -> num:
+    return self.mapped_bytes[k]
+    """
+
+    c = get_contract(code)
+
+    c.set("test", 54321)
+
+    assert c.get("test") == 54321
+
+
+def test_basic_bytes_literal_key():
+    code = """
+mapped_bytes: num[bytes <= 5]
+
+
+def set(v: num):
+    self.mapped_bytes["test"] = v
+
+def get(k: bytes <= 5) -> num:
+    return self.mapped_bytes[k]
+    """
+
+    c = get_contract(code)
+
+    c.set(54321)
+
+    assert c.get("test") == 54321
+
+
+def test_basic_long_bytes_as_keys():
+    code = """
+mapped_bytes: num[bytes <= 34]
+
+def set(k: bytes <= 34, v: num):
+    self.mapped_bytes[k] = v
+
+
+def get(k: bytes <= 34) -> num:
+    return self.mapped_bytes[k]
+    """
+
+    c = get_contract(code)
+
+    c.set("a" * 34, 6789)
+
+    assert c.get("a" * 34) == 6789
+
+
+def test_mismatched_byte_length():
+    code = """
+mapped_bytes: num[bytes <= 34]
+
+def set(k: bytes <= 35, v: num):
+    self.mapped_bytes[k] = v
+    """
+
+    with pytest.raises(TypeMismatchException):
```

Comment:
Can you add a test with a huge number of bytes?

Reply:
@DavidKnott sure thing, uhm what's a huge number ? 4096 bytes?

Reply:
4KB I'd veay is pretty large lol. What's the block storage limit anyways?

Reply:
Average block is like 15KB, so I think 4KB is more than enough